qc_tests/streaks.py

```diff
@@ -203,29 +550,57 @@ def rsc(station, var_list, config_dict, full=False, plots=False, diagnostics=Fal
+        else:
+            wind_speed = None
 
         # need to have at least two observations to get a streak
         if len(obs_var.data.compressed()) >= 2:
-
-            # repeating (straight) strings
             if full:
-                get_repeating_string_threshold(obs_var, config_dict, plots=plots, diagnostics=diagnostics)
-
-            repeating_value(obs_var, station.times, config_dict, plots=plots, diagnostics=diagnostics)
-
-            # more short strings than reasonable
-            # excess_repeating_value()
+                # Recalculating all thresholds for a full run of the QC suite
+
+                # Get thresholds for simple repeating values
+                get_repeating_streak_threshold(obs_var, config_dict, wind_speed=wind_speed,
+                                               plots=plots, diagnostics=diagnostics)
+
+                # Get thresholds for years with excess numbers of streaks
+                get_excess_streak_threshold(obs_var, station.years, config_dict, wind_speed=wind_speed,
+                                            plots=plots, diagnostics=diagnostics)
+
+                # Complete repeat of a day's values
+                # This is slow - as spinning through each day and then selecting
+                repeating_day(obs_var, station, config_dict, determine_threshold=True,
+                              plots=plots, diagnostics=diagnostics)
+
+            # Now run the routines to apply the thresholds and set flags
+                
+            # Simple streaks of repeated values
+            repeating_value(obs_var, station.times, config_dict, wind_speed=wind_speed,
+                            plots=plots, diagnostics=diagnostics)
+            
+            # more short streaks than reasonable            
+            excess_repeating_value(obs_var, station.times, config_dict, wind_speed=wind_speed,
+                                   plots=plots, diagnostics=diagnostics)
 
             # repeats at same hour of day
             # hourly_repeat()
 
             # repeats of whole day
-            # day_repeat()
+            #   (slow)
+            try:
+                # If there is a threshold set, then can use
+                _ = config_dict[f"STREAK-{obs_var.name}"]["DayRepeat"]
+            except KeyError:
+                # if no threshold set, then need to run script to calculate it,
+                #   even if full=False
+                repeating_day(obs_var, station, config_dict, determine_threshold=True, plots=plots, diagnostics=diagnostics)
+            repeating_day(obs_var, station, config_dict, determine_threshold=False, plots=plots, diagnostics=diagnostics)
 
     return # rsc
 
 #************************************************************************
 if __name__ == "__main__":
 
-    print("checking repeated strings")
+    print("checking repeated streaks")
 #************************************************************************
```

Comment:
Correct - it's just to have something there.  You're right, it could be removed or at least the text improved to say what is happening (rather than implying that things are running when they're not!).  I might leave, and raise an issue (#163) so it's done consistently across all of these qc-check files.